### DISCUSSION TITLE
Wish for navigation: option to only start route calculation manually

### DISCUSSION THREAD

**Discussion by felf** [osmandapp/OsmAnd#12579] [Ideas]
Dead devs

More often than not I want to find out a route between two points, but none of which is my current location. Such situations may be to narrow down navigation issues or mapping bugs when I edit osm data. But currently, as soon as I set the destination, Osmand starts calculating the route. If the route is very short, the map view is changed before I can do my next step. And if the route is long, it wastes power. Most of the times, my routes are short, so I need to readjust the map view to find my starting point and set it, which triggers a second route calculation. And then I need to set the navigation mode, which is around 49 % walk, 49 % bicycle and 2 % car (so there is not really a sane default). And then—you guessed it—a third calculation is triggered.

Long story short: May we have an option to disable automatic route calculation and a button „Calculate route“ in the UI, please?

Here’s an example scenario how it may behave:
- The Navigation UI is opened. At the bottom are two buttons: Discard (disabled) and Calculate (enabled if a valid start point is available, such as the current location or a manually placed marker).
- The route points (start, end, intermediate points) are entered and the Calculate button is pressed. Now the route is calculated.
- Once completed, the Discard button is enabled and the Calculate button changes into Start.
- As soon as a route point is changed, the Start button reverts to Calculate.

Thank you very much for your consideration.

**Comment by vshcherb** [comment]
Is it possible to change pattern and start with "Directions from" it doesn't start calculation immediately cause destination is not set
